test/functional/address_types.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test that the wallet can send and receive using all combinations of address types."""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class AddressTypeTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 4
+        self.extra_args = [["-addresstype=legacy"], ["-addresstype=p2sh"], ["-addresstype=p2sh", "-changetype=bech32"], ["-addresstype=bech32"]]
+
+    def run_test(self):
+        self.nodes[0].generate(1)
```

Comment:
Off topic: why is this needed?

Reply:
Otherwise the nodes will think they're in IBD and not sync transactions.